_posts/en/newsletters/2020-02-19-newsletter.md

```diff
@@ -0,0 +1,257 @@
+  contributors.
+
+    1. {:#tap1} Anon asks, "Is Taproot actually more private than bare
+       MAST and schnorr separately?  What are the actual anonymity set
+       benefits compared to doing them separately?"
+
+       Anthony Towns [replies][towns tap], "Yes [it is more private],
+       presuming single-pubkey-single-signature remains a common
+       authorization pattern."  Towns shows that single-sig spends
+       currently represent more than 57% of all transaction outputs (and
+       possibly much more, given the frequent use of P2SH-wrapped
+       P2WPKH).   Either with or without taproot, schnorr will allow
+       extending the set of single-sig spends to cases possible to
+       accomplish with interactive multisig (e.g.  n-of-n), interactive
+       threshold signing (e.g. k-of-n), or adaptor signatures.  Yet only
+       with taproot can the anonymity set also practically extend to
+       cases where spenders prefer to use single-sig but want the option
+       to fall back to using scripts if they can't generate the necessary
+       single signature.  If MAST and schnorr were to be done separately,
+       single-sig users would need to pay more to spend a UTXO than they
+       do now, and so it's unlikely many of them would join the
+       anonymity set that covers users of fallback script.
+
+    2. {:#tap2} Anon asks, "Is Taproot actually cheaper than bare MAST and
+       schnorr separately?"  Earlier in the email, Anon claimed that
+       taproot saves 67 bytes compared to MAST+schnorr for key-path
+       spending but adds 67 bytes for script-path spending.
+
+       Towns points out a redundant data field in Anon's calculation and
+       shows that taproot actually only adds about 33 bytes in the
+       script-path spending case, making the cost-benefit analysis
+       asymmetric in favor of taproot.  David Harding [notes][harding
+       tap] that the extra cost (which translates to 8.25 vbytes) is
+       quite small compared to all the other data a script-path spender
+       would need to provide to spend a UTXO (e.g. 41 vbytes of input
+       data, 16-vbyte signatures or other witnesses of various sizes,
+       one or more 8-vbyte merkle nodes, and the script to execute).
+
+    3. {:#tap3} Anon asks, "Is Taproot riskier than bare MAST and
+       schnorr separately given the new crypto?"
+
+       Towns replies that he "doesn't think so; most of the risk for
+       either of those is in getting the details right. [...] Most of
+       the complicated crypto parts are at the application layer:
+       [MuSig][topic musig], threshold signatures, adaptor signatures,
+       scriptless scripts, etc."  He also links several resources for
+       those wanting to learn more ([1][taplearn1], [2][taplearn2],
+       [3][taplearn3]).
+
+    4. {:#tap4} Anon asks, "couldn't we forego the [Nothing Up My
+       Sleeve] NUMS point requirement and be able to check if it's a
```

Comment:
url?

Reply:
No, the bracketed text is an editorial insertion.  However, a link would probably be a good idea; I'll add that.